monitoring/src/main/scala/akka/viz/aspects/ActorCellInstrumentation.scala

```diff
@@ -21,7 +21,8 @@ class ActorCellInstrumentation {
   def message(jp: JoinPoint, msg: Any, me: ActorCell) {
     if (me.system.name != internalSystemName) {
       Thread.sleep(EventSystem.receiveDelay.toMillis)
-      EventSystem.report(Received(me.sender(), me.self, msg))
+      val canHandle = me.actor.receive.isDefinedAt(msg)
```

Comment:
This won't work, as receive takes into account only static behaviour. Actor can become(behaviour) and I guess for that you would have to instrument Actor.unhandled, or subscribe to UnhandledMessage eventStream